src/component/visualMap/PiecewiseModel.js

```diff
@@ -369,6 +369,30 @@ var PiecewiseModel = VisualMapModel.extend({
             }
         }, this);
 
+        // stop needs to have something
+        if (!stops.length) {
+            // get value axis
+            var cartesian = visualMapModel.ecModel.getComponent('series').coordinateSystem;
+            var baseAxis = cartesian.getBaseAxis();
+            var valueAxis = cartesian.getOtherAxis(baseAxis);
+            var extent = valueAxis.scale.getExtent();
+            zrUtil.each(pieceList, function (piece) {
+                for (var i = 0; i < piece.interval.length; i++) {
+                    // Because CanvasRenderingContext2D doesn't support Infinity
+                    // So set stop value to axis end.
+                    var value = piece.interval[i] === -Infinity
+                        ? extent[0]
+                        : piece.interval[i] === Infinity
+                        ? extent[1]
+                        : piece.interval[i];
+                    stops.push({
+                        value: value,
+                        color: piece.visual.color
+                    });
+                }
+            });
+        }
+
```

Comment:
This specific defensive code should be put in the LineView. https://github.com/apache/incubator-echarts/blob/master/src/chart/line/LineView.js#L157

VisualMap should be not aware of the implementation detail of series.

Reply:
has been a long time (~5 years) since first reported and many duplicates ...
Original PR code would be now at https://github.com/apache/echarts/blob/master/src/component/visualMap/PiecewiseModel.ts#L412.
_"VisualMap is not aware of the implementation detail of series"_ - valid statement, but is [LineView.ts](https://github.com/apache/echarts/blob/master/src/chart/line/LineView.ts) aware of _visualMap_ ?  I see _getVisualGradient_ but it doesn't look like the right place.
Anyway, this code fix is in limbo, not able to find a home...